src/internal/converters/eml/eml_test.go

```diff
@@ -30,11 +33,55 @@ func (suite *EMLUnitSuite) TestConvert_messageble_to_eml() {
 	body, err := os.ReadFile("testdata/email-with-attachments.json")
 	require.NoError(t, err, "reading test file")
 
+	out, err := FromJSON(ctx, body)
+	assert.NoError(t, err, "converting to eml")
+
 	msg, err := api.BytesToMessageable(body)
 	require.NoError(t, err, "creating message")
 
-	_, err = ToEml(ctx, msg)
-	// TODO(meain): add more tests on the generated content
-	// Cannot test output directly as it contains a random boundary
-	assert.NoError(t, err, "converting to eml")
+	assert.Contains(t, out, fmt.Sprintf("Subject: %s", ptr.Val(msg.GetSubject())))
+	assert.Contains(t, out, fmt.Sprintf("Date: %s", msg.GetSentDateTime().Format(time.RFC1123Z)))
+	assert.Contains(
+		t,
+		out,
+		fmt.Sprintf(
+			`From: "%s" <%s>`,
+			ptr.Val(msg.GetFrom().GetEmailAddress().GetName()),
+			ptr.Val(msg.GetFrom().GetEmailAddress().GetAddress())))
+
+	for _, addr := range msg.GetToRecipients() {
+		assert.Contains(
+			t,
+			out,
+			fmt.Sprintf(
+				`To: "%s" <%s>`,
+				ptr.Val(addr.GetEmailAddress().GetName()),
+				ptr.Val(addr.GetEmailAddress().GetAddress())))
+	}
+
+	for _, addr := range msg.GetCcRecipients() {
+		assert.Contains(
+			t,
+			out,
+			fmt.Sprintf(
+				`Cc: "%s" <%s>`,
+				ptr.Val(addr.GetEmailAddress().GetName()),
+				ptr.Val(addr.GetEmailAddress().GetAddress())))
+	}
+
+	for _, addr := range msg.GetBccRecipients() {
+		assert.Contains(
+			t,
+			out,
+			fmt.Sprintf(
+				`Bcc: "%s" <%s>`,
+				ptr.Val(addr.GetEmailAddress().GetName()),
+				ptr.Val(addr.GetEmailAddress().GetAddress())))
+	}
+
+	// Only fist 30 chars as the .eml generator can introduce a
+	// newline in between the text to limit the column width of the
+	// output. It does not affect the data, but can break our tests and
+	// so using 30 as a safe limit to test.
```

Comment:
thanks for the comment update! Is there maybe a little bit more low-hanging fruit around this we can use to check for validity while avoiding flakes? For instance, would something like an assertion that the generated body content length is at least as long as the original input's body content be stable?

Reply:
I think I'll parse the generated `eml` file and test against the parsed data. That should make it really robust.
I'll rework that along with https://github.com/alcionai/corso/pull/4651#discussion_r1394538083